includes/class-wc-payments.php

```diff
@@ -491,15 +491,27 @@ public static function add_plugin_links( $links ) {
 	 * @return array The list of payment gateways that will be available, including WooCommerce Payments' Gateway class.
 	 */
 	public static function register_gateway( $gateways ) {
-		$gateways[] = self::$legacy_card_gateway;
+		$gateways[]         = self::$legacy_card_gateway;
+		$reusable_methods[] = self::$legacy_card_gateway;
 
 		if ( WC_Payments_Features::is_upe_enabled() ) {
 			foreach ( self::$card_gateway->get_payment_method_ids_enabled_at_checkout() as $payment_method_id ) {
 				if ( 'card' === $payment_method_id ) {
 					continue;
 				}
-				$upe_gateway = self::get_payment_gateway_by_id( $payment_method_id );
-				$gateways[]  = $upe_gateway;
+				$upe_gateway        = self::get_payment_gateway_by_id( $payment_method_id );
+				$upe_payment_method = self::get_payment_method_by_id( $payment_method_id );
+
+				if ( $upe_payment_method->is_reusable() ) {
+					$reusable_methods[] = $upe_gateway;
+				}
+
+				$gateways[] = $upe_gateway;
+
+			}
+
+			if ( is_add_payment_method_page() ) {
+				return $reusable_methods;
```

Comment:
I think it makes sense that http://localhost:8082/my-account/add-payment-method/ only display payment methods that are saved, but not every single payment method we support. 👍